cilium/cmd/cleanup.go

```diff
@@ -82,37 +86,47 @@ func runCleanup() {
 		return
 	}
 
+	type cleanupFunc func() error
+	cleanupFuncs := []cleanupFunc{removeAllMaps}
+
+	if !bpfStateOnly {
+		cleanupRoutesAndLinks := func() error {
+			return removeRoutesAndLinks(routes, links)
+		}
+		cleanupNamedNetNSs := func() error {
+			return removeNamedNetNSs(netNSs)
+		}
+
+		cleanupFuncs = append(cleanupFuncs, unmountCgroup, removeDirs, removeCNI,
+			cleanupRoutesAndLinks, cleanupNamedNetNSs)
+	}
+
 	// ENOENT and similar errors are ignored. Should print all other
 	// errors seen, but continue.  So that one remove function does not
 	// prevent the remaining from running.
-	type cleanupFunc func() error
-	checks := []cleanupFunc{removeAllMaps, unmountCgroup, removeDirs, removeCNI}
-	for _, clean := range checks {
+	for _, clean := range cleanupFuncs {
 		if err := clean(); err != nil {
 			fmt.Fprintf(os.Stderr, "Error: %s\n", err)
 		}
 	}
+}
 
-	if err := removeRoutesAndLinks(routes, links); err != nil {
-		fmt.Fprintf(os.Stderr, "Error: %s\n", err)
+func showWhatWillBeRemoved(routes map[int]netlink.Route, links map[int]netlink.Link, netNSs []string) {
+	toBeRemoved := bpfStateToBeRemoved()
+	if !bpfStateOnly {
+		toBeRemoved = append(toBeRemoved, ciliumStateToBeRemoved()...)
 	}
 
-	if err := removeNamedNetNSs(netNSs); err != nil {
-		fmt.Fprintf(os.Stderr, "Error: %s\n", err)
+	warning := "Warning: Destructive operation. You are about to remove:\n"
+	for _, ban := range toBeRemoved {
+		warning += fmt.Sprintf("- %s\n", ban)
 	}
-}
 
-func showWhatWillBeRemoved(routes map[int]netlink.Route, links map[int]netlink.Link, netNSs []string) {
-	fmt.Printf("Warning: Destructive operation. You are about to remove:\n"+
-		"- all BPF maps in %s containing '%s' and '%s'\n"+
-		"- mounted bpffs at %s\n"+
-		"- mounted cgroupv2 at %s\n"+
```

Comment:
what happened to cgroupv2? The new version of this code doesn't seem to specify this, but the function is still invoked.

Reply:
Thanks for catching this, I had missed it. I'll fix in the follow up

Reply:
Fixed!